docs/how-to/debugging-guide.rst

```diff
@@ -0,0 +1,245 @@
+.. meta::
+   :description: A guide to debugging the RCCL library of multi-GPU and multi-node collective communication primitives optimized for AMD GPUs
+   :keywords: RCCL, ROCm, library, API, debug
+
+.. _debugging-rccl:
+
+*********************
+RCCL debugging
+*********************
+
+This topic explains the steps to triage and debug functional and performance issues with RCCL.
+While debugging, collect the output from the commands in this guide for
+support or field personnel. To debug RCCL, follow these steps.
+
+.. _debugging-system-info:
+
+Collecting system information
+=============================
+
+Collect information about the ROCm version, GPU/accelerator, platform, and configuration and provide this
+information to the support team.
+
+*  Verify the ROCm version. This might be a release version or a
+   mainline or staging version. Use this command to display the version:
+
+   .. code:: shell
+
+      cat /opt/rocm/.info/version
+
+*  The problem might be a general issue or specific to the architecture or system.
+   To narrow down the issue, collect information about the GPU or accelerator, along with other
+   details about the platform and system. Some issues to consider include:
+
+   *  Is ROCm running on a bare-metal setup, in a Docker container, in an SR-IOV virtualized
+      environment, or some combination of these configurations? If ROCm is running in a Docker
+      container, provide the name of the Docker image.
+   *  Is the problem only seen on a specific GPU architecture?
+   *  Is it only seen on a specific system type?
+   *  Is it happening on a single node or multinode setup?
+  
+   Run the following command and collect the output:
+
+   .. code:: shell
+
+      rocm_agent_enumerator
+
+   Also collect the name of the GPU or accelerator:
+
+   .. code:: shell
+
+      rocminfo
+
+*  Run the ``rocm-smi`` command to display the system topology.
+
+   .. code:: shell
+
+      rocm-smi
+      rocm-smi --showtopo
+
+*  Determine the values of the ``PATH`` and ``LD_LIBRARY_PATH`` environment variables.
+
+   .. code:: shell
+
+      echo $PATH
+      echo $LD_LIBRARY_PATH
+
+*  Collect the HIP configuration.
+
+   .. code:: shell
+
+      /opt/rocm/bin/hipconfig --full
+
+*  Verify the network settings and setup using the ``ibv_devinfo`` command. 
+   This command displays information about the available RDMA devices and determines 
+   whether they are installed and functioning properly.
+
+   .. code:: shell
+
+      ibv_devinfo
```

Comment:
would be good to include `rdma link`

Reply:
@nileshnegi Can that go into the same section as ibv_devinfo?

Reply:
it's a separate command to run, but yes, it can go in the same section, as it prints a summary of the network links